src/common/snippets/src/op/subgraph.cpp

```diff
@@ -534,54 +539,46 @@ void snippets::op::Subgraph::align_element_types(const BlockedShapeVector& outpu
                                                         return p->get_partial_shape().rbegin()->is_dynamic();
                                                     });
-
-    pre_common.run_passes(body_ptr());
-
-    ov::pass::Manager common_manager;
+    snippets::pass::Manager manager;
     if (config.m_has_domain_sensitive_ops) {
-        common_manager.register_pass<snippets::pass::MatMulToBrgemm>();
-        common_manager.register_pass<snippets::pass::FuseTransposeBrgemm>();
-        common_manager.register_pass<snippets::pass::TransposeDecomposition>();
-        common_manager.register_pass<snippets::pass::SetSoftmaxPorts>();
+        manager.register_pass<snippets::pass::MatMulToBrgemm>();
+        manager.register_pass<snippets::pass::FuseTransposeBrgemm>();
+        manager.register_pass<snippets::pass::TransposeDecomposition>();
+        manager.register_pass<snippets::pass::SetSoftmaxPorts>();
     }
-    common_manager.register_pass<snippets::pass::BroadcastToMoveBroadcast>();
-    common_manager.register_pass<snippets::pass::ConvertConstantsToScalars>();
-    common_manager.register_pass<snippets::pass::ConvertPowerToPowerStatic>();
+    manager.register_pass<snippets::pass::BroadcastToMoveBroadcast>();
+    manager.register_pass<snippets::pass::ConvertConstantsToScalars>();
+    manager.register_pass<snippets::pass::ConvertPowerToPowerStatic>();
     // todo: presently dynamic pipeline is activated even if the last two dimension are static
     //  In general, we can use static kernels in this case, but several parameters (src and dst memory pointers for example)
     //  should be passed as run-time args, so it's a mixed mode: kernel is shape-aware, but some additional runtime args are required
     // Presently Broadcasting is organized in the following way:
     // * ALL last dims are static => broadcasting is handled via MoveBroadcast and pointer arithmetics (even for dynamic upper dims)
     if (!inputs_has_dynamic_last_dims) {
-        common_manager.register_pass<snippets::pass::InsertMoveBroadcast>();
+        manager.register_pass<snippets::pass::InsertMoveBroadcast>();
     }
-    common_manager.run_passes(body_ptr());
-
-    post_common.run_passes(body_ptr());
 
-    ov::pass::Manager precision_manager;
-    precision_manager.register_pass<snippets::pass::PropagatePrecision>(m_generator->get_target_machine());
-    precision_manager.register_pass<ov::pass::ConstantFolding>();
-    precision_manager.register_pass<snippets::pass::ConvertConstantsToScalars>();
-    precision_manager.run_passes(body_ptr());
+    manager.register_pass<snippets::pass::PropagatePrecision>(m_generator->get_target_machine());
+    manager.register_pass<ov::pass::ConstantFolding>();
+    manager.register_pass<snippets::pass::ConvertConstantsToScalars>();
 
-    post_precision.run_passes(body_ptr());
+    for (const auto& pos_pass : backend_passes)
+        manager.register_pass_instance(pos_pass.first, pos_pass.second);
```

Comment:
I think it would be cool method `snippets::pass::Manager::register_positioned_passes` where we hide this logic. What do you think?

Reply:
It's just a few lines, but let's move to a dedicated method, I don't mind.